pkg/controller/certificates/rootcacertpublisher/metrics.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rootcacertpublisher
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+// RootCACertPublisher - subsystem name used by root_ca_cert_publisher
+const RootCACertPublisher = "root_ca_cert_publisher"
+
+var (
+	syncCounter = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      RootCACertPublisher,
+			Name:           "sync_total",
+			Help:           "Number of namespace syncs happened in root ca cert publisher.",
```

Comment:
How/why is this metric useful? Is there an expected "good" range/rate for the values? I'm not sure why a root CA cert publisher would be doing namespace syncs (seems like an implementation detail I wouldn't care about until it breaks :smile:) so it would be nice to either update the help text or add a comment.

Reply:
this is essential for BoundServiceAccountTokenTolume, which can help us troubleshoot without looking into the logs. we hope it can be near 100% :)

Reply:
This is a total counter, split out by label, so how would one derive %? Would that be `metric{code=bad} / sum{metric}`?

Reply:
yes, success ratio: `metric[code=200]/sum`